src/merkle_tree.cpp

```diff
@@ -183,12 +188,12 @@ namespace {
 		}
 
 		// if the mask has only zeros, go straight to empty tree mode
-		if (t.empty() || std::none_of(mask.begin(), mask.end(), identity()))
+		if (t.empty() || std::none_of(mask.begin(), mask.begin() + mask_size, identity()))
```

Comment:
would it make this simpler by truncating `mask` at the beginning, in case it's too long?